### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by karl-von-knecht** [giriss/comfy-image-saver#8]
I love this node but it takes 10x the time as a normal image save. Saving a 4k image on my system (powerful system with a fast SSD) takes 15 seconds. Can you optimize it?

**Comment by green-anger** [comment]
It is slow. I have a somewhat outdated rig (for AI stuff), generation of one image of one of the recommended sizes takes about 30 seconds. When in the same flow I replace the standard saver (or the one from WAS package) with this node it takes more than 50 seconds. As if it's doing something besides saving extra meta data.

**Comment by tusharbhutt** [comment]
I think 'cause it's calculating a hash each time, at least it seems that way in line 218 of the code:

`        modelhash = calculate_sha256(ckpt_path)[:10]`

Also, it's been six weeks since the last update, I think this is not going to be updated :)

**Comment by CHollman82** [comment]
Yes. I removed that and replaced the hash value with 0 and it is much faster now. Of course you don't get the model hash any more... but I'm keeping it this way.

**Comment by alexopus** [comment]
I fixed this in my [fork](https://github.com/alexopus/ComfyUI-Image-Saver) which is available in the Manager as "ComfyUI Image Saver". It will only calculate the hash once per model and store the value on disk, picking that up when needed. Another thing that was slow is the enabled PNG optimization (default comfy doesn't do this), I've made that optional. It made writing PNG ~10x faster for me. Take care if you update, as I renamed the nodes to avoid conflicts with this and added couple of extra widgets.
